docs/user-guide/a1-06-spirv-target-specifics.md

```diff
@@ -0,0 +1,316 @@
+GLSL 4.6 with [GLSL_EXT_shader_atomic_float2](https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GLSL_EXT_shader_atomic_float2.txt) can use atomic operations for 16-bit float type.
+
+SPIR-V 1.5 with [SPV_EXT_shader_atomic_float_add](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_atomic_float_add.asciidoc) and [SPV_EXT_shader_atomic_float_min_max](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_atomic_float_min_max.asciidoc) can use atomic operations for 32-bit float type and 64-bit float type.
+SPIR-V 1.5 with [SPV_EXT_shader_atomic_float16_add](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/EXT/SPV_EXT_shader_atomic_float16_add.asciidoc) can use atomic operations for 16-bit float type
+
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+|        |  32-bit integer | 64-bit integer  |      32-bit float     |  64-bit float    |   16-bit float   |
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+| HLSL   |   Yes (SM5.0)   |   Yes (SM6.6)   | Only bit-wise (SM6.6) |       No         |      No
+| GLSL   |   Yes (GL4.3)   | Yes (GL4.4+ext) |    Yes (GL4.6+ext)    | Yes (GL4.6+ext)  | Yes (GL4.6+ext)
+| SPIR-V |   Yes           |     Yes         |    Yes (SPV1.5+ext)   | Yes (SPV1.5+ext) | Yes (SPV1.5+ext)
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+
+
+ConstantBuffer, (RW)StructuredBuffer (RW)ByteAddressBuffer
+----------------------------------------------------------
+
+Each member in a `ConstantBuffer` will be emitted as `uniform` parameter.
+StructuredBuffer and ByteAddressBuffer are translated to a shader storage buffer with `readonly` layout.
+RWStructuredBuffer and RWByteAddressBuffer are translated to a shader storage buffer with `read-write` layout.
+
+
+ParameterBlock for SPIR-V target
+--------------------------------
+
+`ParameterBlock` is a Slang generic type for binding uniform parameters.
+It is similar to `ConstantBuffer` in HLSL, but `ParameterBlock` can include not only constant parameters but also descriptors such as Texture2D or StructuredBuffer.
+
+Because the Vulkan API doesn't natively support `ParameterBlock` that has a mixture of descriptors and constants, Slang emits each member as an individual parameter.
+When targeting SPIR-V, the individual parameter will be emitted as either `uniform` or `buffer`.
+
+
+SPIR-V specific Compiler options
+--------------------------------
+
+The following compiler options are specific to SPIR-V.
+
+### -emit-spirv-directly
+Generate SPIR-V output directly (default)
+It cannot be used with -emit-spirv-via-glsl
+
+### -emit-spirv-via-glsl
+Generate SPIR-V output by compiling generated GLSL with glslang
```

Comment:
by compiling to glsl source first, then use glslang compiler to produce SPIRV from glsl.

Reply:
Right. The meaning of my writing is not clear enough.
I will change it as suggested.